bulbs/content/models.py

```diff
@@ -337,8 +368,119 @@ def get_serializer_class(cls):
+        # sort via scoring_functions, and then manually filter out zero scores.
+        sponsored_filter = {
+            "query": {
+                "function_score": {
+                    "functions": [
+
+                        # Boost Recent Special Coverage
+                        # Base score is start time
+                        # Note: ES 1.4 sorting granularity is poor for times
+                        # within 1 hour of each other.
+                        {
+                            "field_value_factor": {
+                                "field": "start_date",
+                            }
+                        },
+
+                        # Boost Manually Added Content
+                        {
+                            "filter": {
+                                "terms": {
+                                    "included_ids": [self.id],
+                                }
+                            },
+                            "weight": 10,
+                        },
+                        # Penalize Unsponsored
+                        {
+                            "filter": {
+                                "term": {
+                                    "sponsored": False,
+                                }
+                            },
+                            "weight": 0,
+                        },
+                        # Penalize Inactive
+                        {
+                            "filter": {
+                                "or": [
+                                    {
+                                        "range": {
+                                            "start_date_epoch": {
+                                                "gt": now_epoch,
+                                            },
+                                        }
+                                    },
+                                    {
+                                        "range": {
+                                            "end_date_epoch": {
+                                                "lte": now_epoch,
+                                            },
+                                        }
+                                    },
+                                ],
+                            },
+                            "weight": 0,
+                        },
+                    ],
+                },
+            },
+
+            "sort": "_score",  # The only sort method supported by ES v1.4 percolator
+            "size": max_size,  # Required for sort
+        }
+
+        results = _percolate(index=self.mapping.index,
+                             doc_type=self.mapping.doc_type,
+                             content_id=self.id,
+                             body=sponsored_filter)
+
+        return [r["_id"] for r in results
+                # Zero score used to omit results via scoring function (ex: unsponsored)
+                if r['_score'] > 0]
```

Comment:
We use unsponsored special coverage now - just want to make sure those aren't going to be missed with this if that score is 0 as well as the inactive ones